mypy/checkmember.py

```diff
@@ -797,17 +804,35 @@ def type_object_type(info: TypeInfo, builtin_type: Callable[[str], Instance]) ->
         # Otherwise prefer __init__ in a tie. It isn't clear that this
         # is the right thing, but __new__ caused problems with
         # typeshed (#5647).
-        method = init_method
+        method = init_method.node
     # Construct callable type based on signature of __init__. Adjust
     # return type and insert type arguments.
-    return type_object_type_from_function(method, info, fallback)
+    if isinstance(method, FuncBase):
+        t = function_type(method, fallback)
+    else:
+        assert isinstance(method.type, FunctionLike)  # is_valid_constructor() ensures this
+        t = method.type
+    signature = bind_self(t)
```

Comment:
Any particular reason for moving this `bind_self` out of `type_object_type_from_function`? I don't have any objection, I'm just wondering if there is a technical reason I'm missing or you just like it more.

Reply:
No particular reason. I think I started refactoring by pulling `function_type()` out of there, and pulled whole line out of there. But looking again at it I think it makes sense to keep bind and map together, so I will put it back.